README.md
# gavin-voice-vscode README

This is the README for your extension "gavin-voice-vscode". After writing up a brief description, we recommend including the following sections.

## Features

Describe specific features of your extension including screenshots of your extension in action. Image paths are relative to this README file.

For example if there is an image subfolder under your extension project workspace:

\!\[feature X\]\(images/feature-x.png\)

> Tip: Many popular extensions utilize animations. This is an excellent way to show off your extension! We recommend short, focused animations that are easy to follow.

## Requirements

If you have any requirements or dependencies, add a section describing those and how to install and configure them.

## Extension Settings

Include if your extension adds any VS Code settings through the `contributes.configuration` extension point.

For example:

This extension contributes the following settings:

* `myExtension.enable`: Enable/disable this extension.
* `myExtension.thing`: Set to `blah` to do something.

## Known Issues

Calling out known issues can help limit users opening duplicate issues against your extension.

## Release Notes

Users appreciate release notes as you update your extension.

### 1.0.0

Initial release of ...

### 1.0.1

Fixed issue #.

### 1.1.0

Added features X, Y, and Z.

---

## Following extension guidelines

Ensure that you've read through the extensions guidelines and follow the best practices for creating your extension.

* [Extension Guidelines](https://code.visualstudio.com/api/references/extension-guidelines)

## Working with Markdown

You can author your README using Visual Studio Code. Here are some useful editor keyboard shortcuts:

* Split the editor (`Cmd+\` on macOS or `Ctrl+\` on Windows and Linux).
* Toggle preview (`Shift+Cmd+V` on macOS or `Shift+Ctrl+V` on Windows and Linux).
* Press `Ctrl+Space` (Windows, Linux, macOS) to see a list of Markdown snippets.

## For more information

* [Visual Studio Code's Markdown Support](http://code.visualstudio.com/docs/languages/markdown)
* [Markdown Syntax Reference](https://help.github.com/articles/markdown-basics/)

**Enjoy!***


CHANGELOG.md
# Change Log

All notable changes to the "gavin-voice-vscode" extension will be documented in this file.

Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.

## [Unreleased]

- Initial release

src/extension.ts
// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
import * as vscode from 'vscode';
const speech = require('@google-cloud/speech');
const recorder = require('node-record-lpcm16');
const client = new speech.SpeechClient();


// This method is called when your extension is activated
// Your extension is activated the very first time the command is executed
export function activate(context: vscode.ExtensionContext) {
	//demander à l'utilisateur si vscode peut accéder à son micro
	//ecrire dans la console de vscode pour dire que l'extension est activée
	let outputChannel = vscode.window.createOutputChannel("Mon Extension");
	outputChannel.appendLine("Hello World!");
	outputChannel.show();
	vscode.window.showInformationMessage('Do you want to allow VSCode to access your microphone?', 'Yes', 'No').then((answer) => {
		if (answer === 'Yes') {
			//si l'utilisateur accepte, on lance la fonction qui va écouter le micro
			//startListening();
		}
	});


	
	/**
	 * TODO(developer): Uncomment the following lines before running the sample.
	 */
	 const encoding = 'Encoding of the audio file, e.g. LINEAR16';
	 const sampleRateHertz = 16000;
	 const languageCode = 'BCP-47 language code, e.g. en-US';
	
	const request = {
	  config: {
		encoding: encoding,
		sampleRateHertz: sampleRateHertz,
		languageCode: languageCode,
	  },
	  interimResults: false, // If you want interim results, set this to true
	};
	
	// Create a recognize stream
	const recognizeStream = client
	  .streamingRecognize(request)
	  .on('error', console.error)
	  .on('data',(data: any) =>
		process.stdout.write(
		  data.results[0] && data.results[0].alternatives[0]
			? `Transcription: ${data.results[0].alternatives[0].transcript}\n`
			: '\n\nReached transcription time limit, press Ctrl+C\n'
		)
	  );
	
	// Start recording and send the microphone input to the Speech API.
	// Ensure SoX is installed, see https://www.npmjs.com/package/node-record-lpcm16#dependencies
	recorder
	  .record({
		sampleRateHertz: sampleRateHertz,
		threshold: 0,
		// Other options, see https://www.npmjs.com/package/node-record-lpcm16#options
		verbose: false,
		recordProgram: 'rec', // Try also "arecord" or "sox"
		silence: '10.0',
	  })
	  .stream()
	  .on('error', console.error)
	  .pipe(recognizeStream);
	
	console.log('Listening, press Ctrl+C to stop.');

	
}

// This method is called when your extension is deactivated
export function deactivate() {}


src/extension.js
"use strict";
exports.__esModule = true;
exports.deactivate = exports.activate = void 0;
// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
var vscode = require("vscode");
var speech = require('@google-cloud/speech');
var recorder = require('node-record-lpcm16');
var client = new speech.SpeechClient();
// This method is called when your extension is activated
// Your extension is activated the very first time the command is executed
function activate(context) {
    //demander à l'utilisateur si vscode peut accéder à son micro
    vscode.window.showInformationMessage('Do you want to allow VSCode to access your microphone?', 'Yes', 'No').then(function (answer) {
        if (answer === 'Yes') {
            //si l'utilisateur accepte, on lance la fonction qui va écouter le micro
            //startListening();
        }
    });
    /**
     * TODO(developer): Uncomment the following lines before running the sample.
     */
    var encoding = 'Encoding of the audio file, e.g. LINEAR16';
    var sampleRateHertz = 16000;
    var languageCode = 'BCP-47 language code, e.g. en-US';
    var request = {
        config: {
            encoding: encoding,
            sampleRateHertz: sampleRateHertz,
            languageCode: languageCode
        },
        interimResults: false
    };
    // Create a recognize stream
    var recognizeStream = client
        .streamingRecognize(request)
        .on('error', console.error)
        .on('data', function (data) {
        return process.stdout.write(data.results[0] && data.results[0].alternatives[0]
            ? "Transcription: ".concat(data.results[0].alternatives[0].transcript, "\n")
            : '\n\nReached transcription time limit, press Ctrl+C\n');
    });
    // Start recording and send the microphone input to the Speech API.
    // Ensure SoX is installed, see https://www.npmjs.com/package/node-record-lpcm16#dependencies
    recorder
        .record({
        sampleRateHertz: sampleRateHertz,
        threshold: 0,
        // Other options, see https://www.npmjs.com/package/node-record-lpcm16#options
        verbose: false,
        recordProgram: 'rec',
        silence: '10.0'
    })
        .stream()
        .on('error', console.error)
        .pipe(recognizeStream);
    console.log('Listening, press Ctrl+C to stop.');
}
exports.activate = activate;
// This method is called when your extension is deactivated
function deactivate() { }
exports.deactivate = deactivate;
